clippy_lints/src/methods/mod.rs

```diff
@@ -2052,43 +2135,18 @@ fn lint_filter_map_map<'a, 'tcx>(
     _filter_args: &'tcx [hir::Expr],
     _map_args: &'tcx [hir::Expr],
 ) {
-    // lint if caller of `.filter().map()` is an Iterator
-    if match_trait_method(cx, expr, &paths::ITERATOR) {
-        let msg = "called `filter_map(p).map(q)` on an `Iterator`. \
-                   This is more succinctly expressed by only calling `.filter_map(..)` instead.";
-        span_lint(cx, FILTER_MAP, expr.span, msg);
-    }
-}
-
-/// lint use of `filter().flat_map()` for `Iterators`
-fn lint_filter_flat_map<'a, 'tcx>(
-    cx: &LateContext<'a, 'tcx>,
-    expr: &'tcx hir::Expr,
-    _filter_args: &'tcx [hir::Expr],
-    _map_args: &'tcx [hir::Expr],
-) {
-    // lint if caller of `.filter().flat_map()` is an Iterator
-    if match_trait_method(cx, expr, &paths::ITERATOR) {
-        let msg = "called `filter(p).flat_map(q)` on an `Iterator`. \
-                   This is more succinctly expressed by calling `.flat_map(..)` \
-                   and filtering by returning an empty Iterator.";
-        span_lint(cx, FILTER_MAP, expr.span, msg);
+    let mut span = expr.span;
+    if let hir::ExprKind::MethodCall(_, _, ref expr_vec) = expr.node {
+        if let hir::ExprKind::MethodCall(_, ref filter_span, _) = expr_vec[0].node {
+            span = Span::new(filter_span.lo(), expr.span.hi(), expr.span.ctxt());
```

Comment:
There's already a function for this on `Span`: https://doc.rust-lang.org/nightly/nightly-rustc/src/syntax_pos/lib.rs.html#217-219

```suggestion
            span = expr.span.with_lo(filter_span.lo());
```